### PR TITLE
wiso-steuer-2024 31.09.4100

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by p-linnane** [comment]
This PR addresses the issue raised in https://github.com/orgs/Homebrew/discussions/5548.

File: Casks/w/wiso-steuer-2024.rb [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
   name "WISO Steuer 2024"
   desc "Tax declaration for the fiscal year 2023"
   homepage "https://www.buhl.de/download/wiso-steuer-2024/"
 
   livecheck do
     url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Aktuell/appcast-steuer.xml"
-    strategy :sparkle
+    regex(%r{/Files/(\d+(?:\.\d+)+)(-\w+\d)/SteuerMac2024[._-]v?(\d+(?:\.\d+)+).dmg}i)
+    strategy :sparkle do |item, regex|
+      match = item.url.match(regex)
```
Comment (khipp) [inline Casks/w/wiso-steuer-2024.rb:16]:
```suggestion
      match = item.url&.match(regex)
```

I'm unsure if the implementation allows for `url` to be `nil`, but it wouldn't hurt to be on the safe side.

File: Casks/w/wiso-steuer-2024.rb [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+  url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Files/#{version.csv.first}#{version.csv.second}/SteuerMac2024-#{version.csv.first}.dmg",
       verified: "update.buhl-data.com/Updates/Steuer/"
   name "WISO Steuer 2024"
   desc "Tax declaration for the fiscal year 2023"
   homepage "https://www.buhl.de/download/wiso-steuer-2024/"
 
   livecheck do
     url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Aktuell/appcast-steuer.xml"
-    strategy :sparkle
+    regex(%r{/Files/(\d+(?:\.\d+)+)(-\w+\d)/SteuerMac2024[._-]v?(\d+(?:\.\d+)+).dmg}i)
```
Comment (khipp) [inline Casks/w/wiso-steuer-2024.rb:14]:
```suggestion
    regex(%r{/Files/(\d+(?:\.\d+)+)([_-]\w+\d+)/SteuerMac2024[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
```

These are merely suggestions to account for slight changes in the naming scheme.

**Review by khipp — COMMENTED** [review]
We probably should also update the related casks [wiso-steuer-2022](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/w/wiso-steuer-2022.rb) and [wiso-steuer-2023](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/w/wiso-steuer-2023.rb).

File: Casks/w/wiso-steuer-2024.rb [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
 
   livecheck do
     url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Aktuell/appcast-steuer.xml"
-    strategy :sparkle
+    regex(%r{/Files/(\d+(?:\.\d+)+)([_-]\w+\d+)/SteuerMac2024[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
+    strategy :sparkle do |item, regex|
+      match = item.url&.match(regex)
+      next if match.blank?
+
+      "#{match[1]},#{match[2]}"
```
Comment (samford) [inline Casks/w/wiso-steuer-2024.rb:19]:
```suggestion
    regex(%r{/v?(\d+(?:\.\d+)+)([^/]+)?/SteuerMac2024[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
    strategy :sparkle do |item, regex|
      match = item.url&.match(regex)
      next if match.blank?

      match[2] ? "#{match[1]},#{match[2]}" : match[1]
```

The unstable suffix (`-RC1`) won't always be present, so we have to make it optional and handle the presence/absence in the regex/`strategy` block. The existing `url` approach in this PR will work with both, though.

**Review by samford — COMMENTED** [review]
Outside of the suggestion below, one other way to handle this would be to match the full version from the Sparkle URL (`31.09.4100-RC1`) and then use the part of the version before the suffix in the `url` filename, like so:

```diff
diff --git a/Casks/w/wiso-steuer-2024.rb b/Casks/w/wiso-steuer-2024.rb
index cc83f443fd0bdec24315fc9b1e72c83cedb87be7..d2fbcd32ef3f0a2c8e85f6bb9b5c9552acfcb68b 100644
--- a/Casks/w/wiso-steuer-2024.rb
+++ b/Casks/w/wiso-steuer-2024.rb
@@ -1,9 +1,9 @@
 cask "wiso-steuer-2024" do
   # NOTE: "2024" is not a version number, but an intrinsic part of the product name
-  version "31.09.4100,-RC1"
+  version "31.09.4100-RC1"
   sha256 "3b21d169989dd7a799c318879839a6f2bfb0e0703021f13a5e9cba3c3d32d897"
 
-  url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Files/#{version.csv.first}#{version.csv.second}/SteuerMac2024-#{version.csv.first}.dmg",
+  url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Files/#{version}/SteuerMac2024-#{version.split("-").first}.dmg",
       verified: "update.buhl-data.com/Updates/Steuer/"
   name "WISO Steuer 2024"
   desc "Tax declaration for the fiscal year 2023"
@@ -11,12 +11,12 @@ cask "wiso-steuer-2024" do
 
   livecheck do
     url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Aktuell/appcast-steuer.xml"
-    regex(%r{/Files/(\d+(?:\.\d+)+)([_-]\w+\d+)/SteuerMac2024[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
+    regex(%r{/v?(\d+(?:\.\d+)+[^/]*)/SteuerMac2024[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
     strategy :sparkle do |item, regex|
       match = item.url&.match(regex)
       next if match.blank?
 
-      "#{match[1]},#{match[2]}"
+      match[1]
     end
   end
```

If that's preferable, I can push a commit. Otherwise, the `livecheck` block suggestion below will work.

**Comment by p-linnane** [comment]
@samford I defer to you on what you think it best. Just want to make sure we have a (reasonably) future proof solution in place.

**Comment by samford** [comment]
> I defer to you on what you think it best. Just want to make sure we have a (reasonably) future proof solution in place.

Looking around, using the full version without any commas maybe feels a bit more in line with most other casks when it comes to unstable suffixes like `-RC1`. There appears to be a similar `version`/`url` approach in `remoteviewer`, so there's some level of prior art for the `version.split("-").first` idea. I'll push a commit in a moment.

**Comment by p-linnane** [comment]
Thanks as always @samford 🙏

**Review by khipp — APPROVED** [review]

**Pull request merged**